### DISCUSSION TITLE
Edge cases not supported by moPepGen

### DISCUSSION THREAD

**Discussion by zhuchcn** [uclahs-cds/package-moPepGen#472] [General]
Edge cases below are skipped by moPepGen callVariant for technical difficulties:

- Any variants that overlaps with the start codon for coding transcripts, and first three nucleotide for noncoding transcripts. 
- Any variants that overlaps with the splicing site
- Fusion that the donor breakpoint is smaller than 3 at the donor transcript. #633 
- Any variant that overlaps with the last nucleotide of transcripts with the `mrna_end_NF` tag (meaning that the end of the transcript is not clear).

**Comment by lydiayliu** [comment]
There are 39 peptides only called when grouping SNV and INDEL together, and 14 only called when SNV and INDEL are parsed separately by parseVEP, and they are all noncoding. My thought is this is not a problem of calling parseVEP differently, but is in general that callVariant is a little unstable on noncoding transcripts, that some edge cases are not handled properly. So that means even if you run callVariant on the same GVF twice, you may get slightly different results. So some peptides are called and some are not called from noncoding transcripts. We know this because fuzzTest also told us that noncoding transcripts still have some mismatches.

_Originally posted by @zhuchcn in https://github.com/uclahs-cds/project-MissingPeptides-Method/issues/124#issuecomment-1204620262_
